src/modules/voxelDensity/process.cpp

```diff
@@ -0,0 +1,112 @@
+#include "analyser/dataExporter.h"
+#include "analyser/dataOperator1D.h"
+#include "main/dissolve.h"
+#include "math/gaussFit.h"
+#include "module/context.h"
+#include "voxelDensity.h"
+
+void VoxelDensityModule::addValue(Vec3<double> coords, double value, Array3D<double> &array)
+{
+    if (nAxisVoxels_.x == 1 && nAxisVoxels_.y == 1 && nAxisVoxels_.z == 1)
+    {
+        array.values()[0] += value;
+        return;
+    }
+
+    auto t = std::make_tuple((int)std::floor(coords.x * nAxisVoxels_.x), (int)std::floor(coords.y * nAxisVoxels_.y),
+                             (int)std::floor(coords.z * nAxisVoxels_.z));
+    array[t] += value;
+}
+
+Vec3<double> VoxelDensityModule::foldedCoordinates(const Vec3<double> &r, const Box *unitCell) { return unitCell->foldFrac(r); }
+
+Module::ExecutionResult VoxelDensityModule::process(ModuleContext &context)
+{
+    auto &processingData = context.dissolve().processingModuleData();
+
+    // Define voxels
+    auto unitCell = targetConfiguration_->box();
+    const double boxX = unitCell->axisLengths().x, boxY = unitCell->axisLengths().y, boxZ = unitCell->axisLengths().z;
+    voxelVolume_ = voxelSideLength(boxX) * voxelSideLength(boxY) * voxelSideLength(boxZ);
+    nAxisVoxels_.x = int(round(boxX / voxelSideLength(boxX)));
+    nAxisVoxels_.y = int(round(boxY / voxelSideLength(boxY)));
+    nAxisVoxels_.z = int(round(boxZ / voxelSideLength(boxZ)));
+
+    Messenger::print("Volume of unit cell voxels: {}.\n", voxelVolume_);
+    Messenger::print("Number of voxels along each axis: nX={}, nY={}, nZ={}.\n", nAxisVoxels_.x, nAxisVoxels_.y,
+                     nAxisVoxels_.z);
+
+    // Calculate target property 3d map over unit cell voxels
+    auto array3D = processingData.realise<Array3D<double>>("Array3D", name());
+    array3D.initialise(nAxisVoxels_.x, nAxisVoxels_.y, nAxisVoxels_.z);
+
+    if (!restrictToSpecies_.empty())
+    {
+        auto cfgSpecies = targetConfiguration_->speciesPopulations();
+
+        for (const auto &[sp, N] : cfgSpecies)
+        {
+            bool included = false;
+
+            for (const auto &spTarget : restrictToSpecies_)
+            {
+                if (sp->name() == spTarget->name())
+                {
+                    included = true;
+                    break;
+                }
+            }
```

Comment:
```suggestion
            bool included = std::any_oF(restricToToSpecies_.begin(), restrictToSpecies_.end(), [sp](auto &spTarget) {return sp->name() == spTarget->name();});
```

Might be a good place to use `std::any_of`

Reply:
Agreed